doc/devel/contributing.rst

```diff
@@ -21,44 +21,63 @@ also welcome to post feature requests or pull requests.
 Retrieving and installing the latest version of the code
 ========================================================
 
+When working on the Matplotlib source, setting up a `virtual
+environment
+<http://docs.python-guide.org/en/latest/dev/virtualenvs/>`_ or a
+`conda environment <http://conda.pydata.org/docs/using/envs.html>`_ is
+recommended.
+
+.. warning::
+
+   If you already have a version of Matplotlib installed, you will need to
+   uninstall it.
```

Comment:
Only in the same venv, I think?  AFAICT that looks big and scary while it is simply not true if the instructions above are followed :-)